### ISSUE TITLE
feat(Reddit): add `Change package name` patch

### ISSUE THREAD

**Issue by nikhilbadyal** [inotia00/ReVanced_Extended#1842]
### Application

Reddit

### Feature description

Google made some in recent days and now [modded apps also show up in Play Store ](https://www.reddit.com/r/revancedapp/comments/18i2r1z/did_google_change_something_from_their_end_as_of/)Update menu. This is creating an issue for[ reddit app](https://www.reddit.com/r/Piracy/comments/18i7696/apps_not_downloaded_from_the_play_store_trying_to/). 

Can we migrate [this](https://revanced.app/patches?s=Change+Package+Name) patch from Revanced to change package for Reddit Extended? Or a new patch for Reddit also like YT & YT Music. 

### Motivation

After being shown in Play Store it's annoying and wastes bandwidth as even if we click on update in Playstore. It downloads the update and fails in the end.

### Acknowledgements

- [X] This is not a feature request for YouTube.
- [X] This issue is not a duplicate of an existing feature request.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
I haven't tested that patch
iirc in certain apps, force close occurred when changing the package name

**Comment by nikhilbadyal** [comment]
It works for reddit atleast ig. As per reddit discussions.

**Comment by inotia00** [comment]
I have finished testing to implement this patch.

I was able to implement the patch in two ways, and Pros and Cons according to each method are as follows.

**Method 1. The 'Change package name' patch provided by the official ReVanced Patches**
: This only changes AndroidManifest's 'package' name.

- **Pros**: Available in RVX Manager and CLI.
- **Cons**: Unlike YouTube or YT Music, you could not install multiple Reddit apps.

**Method 2. The 'Change package name' patch that has been improved a bit on my side**
: Change not only AndroidManifest's 'package' but also the permissions and provider's package name.

- **Pros**: You can install multiple Reddit apps, such as YouTube or YT Music.
- **Cons**: Not available in RVX Manager.

I decided to combine Method 1 and Method 2.

In other words, if you try patching in **RVX Manager**, Method 1 will be applied.
If you try patching in **Revancify, rvx-builder, or CLI**, Method 2 will be applied.

This will be reflected in the next release.

**Comment by nikhilbadyal** [comment]
I always use cli. Anyways good that you decided to go ahead with 1 & 2 both.

**Comment by inotia00** [comment]
reflected in [revanced-patches-v2.207.1](https://github.com/inotia00/revanced-patches/releases/tag/v2.207.1)
